miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppSecureStorageDispatcher.kt

```diff
@@ -193,47 +145,77 @@ internal class MiniAppSecureStorageDispatcher(
     }
 
     fun onClearAll(callbackId: String) = whenReady {
-        onSuccessClearSecureStorage = {
-            cachedItems = null
+        onSuccess = {
             bridgeExecutor.postValue(callbackId, REMOVE_SUCCESS_SECURE_STORAGE)
         }
         onFailed = { errorSecure: MiniAppSecureStorageError ->
             bridgeExecutor.postError(callbackId, Gson().toJson(errorSecure))
         }
-        secureStorage.delete(miniAppId, onSuccessClearSecureStorage, onFailed)
+        miniAppSecureStorage.delete(onSuccess, onFailed)
     }
 
     @Suppress("MagicNumber")
+    @Deprecated("No Longer Needed")
     fun onSize(callbackId: String) = whenReady {
-        onSuccessFileSize = { fileSize: Long ->
+        onSuccessDBSize = { fileSize: Long ->
             val maxSizeInBytes = storageMaxSizeKB * 1024
             val storageSize =
                 Gson().toJson(MiniAppSecureStorageSize(fileSize, maxSizeInBytes.toLong()))
             bridgeExecutor.postValue(callbackId, storageSize)
         }
-        secureStorage.secureStorageSize(miniAppId, onSuccessFileSize)
+        miniAppSecureStorage.getDatabaseUsedSize(miniAppId, onSuccessDBSize)
     }
 
-    fun cleanupSecureStorage() {
-        cachedItems = null
-        secureStorage.storageState.removeObserver(stateObserver)
-    }
+    fun cleanupSecureStorage() {}
 
     /**
      * Will be invoked by MiniApp.clearSecureStorage(miniAppId: String).
      * @param miniAppId will be used to find the storage to be deleted.
      */
-    fun clearSecureStorage(miniAppId: String) = secureStorage.clearSecureStorage(miniAppId)
+    fun clearSecureStorage(miniAppId: String) = whenReady {
+        clearSecureDatabase(miniAppId)
+    }
+
+    /**
+     * Will be invoked by MiniApp.clearSecureStorage.
+     */
+    fun clearSecureStorage() = whenReady {
+        clearAllSecureDatabases()
+    }
+
+    /**
+     * It will delete all the records as well as the whole DB related to the given mini app id
+     * Will be invoked with MiniApp.clearSecureStorage(miniAppId: String).
+     * @param miniAppId will be used to find the file to be deleted.
+     */
+    private fun clearSecureDatabase(miniAppId: String) {
+        try {
+            val dbName = DB_NAME_PREFIX + miniAppId
+            activity.deleteDatabase(dbName)
+        } catch (e: Exception) {
+            // No callback needed. So Ignoring.
+        }
+    }
 
     /**
      * Will be invoked by MiniApp.clearSecureStorage.
      */
-    fun clearSecureStorage() = secureStorage.clearSecureStorage()
+    private fun clearAllSecureDatabases() {
+        try {
+            activity.databaseList().forEach {
+                if (it.startsWith(DB_NAME_PREFIX)) {
+                    activity.deleteDatabase(it)
+                }
+            }
+        } catch (e: Exception) {
+            // No callback needed. So Ignoring.
```

Comment:
also here

Reply:
Yeah, This one can be removed as I was just trying to be extra cautious.